README.md

```diff
@@ -18,6 +18,90 @@ to facilitate unit-testing of OpenTracing Java instrumentation.
+ 
+It's inconvenient to pass an active `Span` from function to function manually, so OpenTracing provides a `ActiveSpanHolder` abstraction to provide access to the active `Span` and to capture it for reactivation in other execution contexts (e.g., in an async callback).
+
+Every `Tracer` implementation _must_ provide access to a `ActiveSpanHolder` (typically provided at `Tracer` initialization time). The `ActiveSpanHolder` in turn exposes the active `Span`, like so:
+
+```
+    io.opentracing.Tracer tracer = ...;
+    ...
+    Span active = tracer.holder().activeSpan();
+    if (active != null) {
+        active.log("...");
+    }
+```
+
+### Starting a new Span
+
+The simplest case looks like this:
+
+```
+    io.opentracing.Tracer tracer = ...;
+    ...
+    try (Span span = tracer.buildSpan("someWork").start()) {
+        // (do things / record data to `span`)
+    }
+```
+
+Or, to take advantage of `ActiveSpanHoldar` and automatic intra-process propagation, like this:
+
+```
+    io.opentracing.Tracer tracer = ...;
+    ...
+    try (ActiveSpanHoldar.Continuation cont = tracer.buildSpan("someWork").startAndActivate()) {
+        Span span = cont.span();
+        // (do things / record data to `span`)
+    }
+```
+
+
+**If there is an active `Span` per `ActiveSpanHoldar`, it will act as the parent to any newly started `Span`** unless the programmer provides an explicit reference at `buildSpan` time, like so:
+
+```
+    io.opentracing.Tracer tracer = ...;
+    ...
+    Span span = tracer.buildSpan("someWork").asChildOf(otherSpanContext).start();
+```
+
+### Deferring asynchronous work
+
+Consider the case where a `Span`'s lifetime logically starts in one execution context and ends in another. For instance, the intra-Span timing breakdown might look like this:
+
+```
+[ ServiceHandlerSpan                                 ]
+|-FunctionA-|-----waiting on an RPC------|-FunctionB-|
+            
+------------------------------------------------> time
+```
+
+The `ServiceHandlerSpan` is _active_ when it's running FunctionA and FunctionB, and inactive while it's waiting on an RPC (presumably modelled as its own Span, though that's not the concern here).
+
+**The `ActiveSpanHolder` makes it easy to capture the Span and execution context in `FunctionA` and re-activate it in `FunctionB`.** These are the steps:
+
+1. In the method/function that *allocates* the closure/`Runnable`/`Future`/etc, call `ActiveSpanHolder#captureActive()` to obtain a `ActiveSpanHolder.Continuation`
+2. In the closure/`Runnable`/`Future`/etc itself, pair calls to `ActiveSpanHolder.Continuation#activate` and `ActiveSpanHolder.Continuation#deactivate`
+
+In practice, the latter is most fluently accomplished through the use of an OpenTracing-aware `ExecutorService` and/or `Runnable`/`Callable` adapter; they can factor all of the above.
```

Comment:
might be useful to have an actual code example, as this is arguably one of the most critical (difficult to get right) use cases.